cosmic_ray/parsing.py

```diff
@@ -22,15 +23,21 @@ def get_ast(module):
         TypeError: If the source file for `module` can't be found.
     """
     try:
-        source = inspect.getsource(module)
-    except OSError:
-        LOG.warning("Unable to read source for module %s", module)
-        raise
-
-    try:
         source_file = inspect.getsourcefile(module)
     except TypeError:
+        source_file = ''
         LOG.error("Unable to get source file for object %s", module)
         raise
 
+    try:
+        source = inspect.getsource(module)
+    except OSError:
+        # workaround for empty __init__.py files, see
+        # http://bugs.python.org/issue27578
+        if source_file.endswith('__init__.py') and os.path.exists(source_file):
+            source = open(source_file, 'r').read()
```

Comment:
Won't `source` always be an empty string in this case? If so, we don't need to open and read the file.